### PR TITLE
Add support for custom handler in the _handleEvent function

### PR DESCRIPTION
**Pull request by HerbertVidela**
Changes made:
- Added the new prop onAction to handle the case when you want add a custom functionality after the user do some  action.
- Fix the `reaming` validation in the _handleEvent function, if it's zero then the event won't be skipped.

### CODE REVIEW

**Comment by SupremeTechnopriest** [comment]
Hey @HerbertVidela!  Thanks for your feature submission.  I took a look and short of a few clean ups I'm down to merge this in. Could you explain your use case to me?  Wouldn't you want this kind of functionality outside of IdleTimer?  This event will fire on ANY action. Including every keystroke and scroll event.

**Comment by HerbertVidela** [comment]
Hi @SupremeTechnopriest,

**TL;DR**. My use case is I need to save a value after an action done by the user. And a user of the library can debounce the function pass to the onAction prop from outside the component.

Well my use case is kind of special, I need to logout a user after X minutes which, alone, could be solve by the current implementation. But I need to take into consideration that the user can have multiple tabs open with the app and if he continue doing actions in an "active" tab then he shouldn't be logout if one of the timers in the other "inactive" tabs hits zero, to solve that I'll need to save a the date of the last activity done by the user after an action.

About the second question, I wonder if the _eventHandler should have a debounce function and expose the timeout for each one, but I didn't want to add to changes for something I'm not sure about, so what I think is that the user can debounce the function passed to the prop onAction and left the current implementation as it is.

**Comment by SupremeTechnopriest** [comment]
Hey @HerbertVidela, sorry for the delay on this.  We are getting the feature added in #62.  Thanks for your work here.  It's basically identical to what you have here.  Look out for the next release!

**Comment by SupremeTechnopriest** [comment]
Hey @HerbertVidela! Version `4.1.0` is on npm and contains this feature.  Enjoy!
